core/src/actors/paths.rs

```diff
@@ -292,22 +305,42 @@ impl ActorPath {
         from.dispatcher_ref().enqueue(MsgEnvelope::Typed(env))
     }
 
-    /// Same as [tell](ActorPath::tell), but serialises eagerly
-    pub fn tell_ser<S, B, E>(&self, m: B, from: &S) -> Result<(), E>
-    where
-        S: ActorSource,
-        B: TryInto<Serialised, Error = E>,
+    /// Same as [tell](ActorPath::tell), but serialises eagerly into a Pooled buffer (pre-allocated and bounded)
+    pub fn tell_serialised<CD, B>(&self, m: B, from: &CD) -> Result<(), SerError>
+        where
+            CD: ComponentDefinition + Sized + 'static,
+            B: Serialisable + 'static,
     {
-        let msg: Serialised = m.try_into()?;
-        let src = from.path_resolvable();
-        let dst = self.clone();
-        let env = DispatchEnvelope::Msg {
-            src,
-            dst,
-            msg: DispatchData::Eager(msg),
-        };
-        from.dispatcher_ref().enqueue(MsgEnvelope::Typed(env));
-        Ok(())
+        if self.protocol() == Transport::LOCAL {
+            // No need to serialize!
+            self.tell(m, from);
+            return Ok(());
+        }
+        //let mut buf = encode_buffer.deref_mut();
+        let mut buf_ref = from.ctx().get_buffer().borrow_mut();
+        let buf = buf_ref.deref_mut();
+        // Reserve space for the header:
+        buf.pad(FRAME_HEAD_LEN as usize);
+
+        from.actor_path().serialise(buf)?; // src
+        self.clone().serialise(buf)?; // dst
+        buf.put_ser_id(m.ser_id()); // ser_id
```

Comment:
Things are put into the buffer here. What would happen if the following call fails and tell_serialised is called again?
```rust
buf.get_chunk()
```
It would pad and write the ser_id into the buffer once again, right?  Or is this handled somewhere?

Reply:
It takes new buffer every time, though, no?

Reply:
ah, right. the EncodeBuffer uses a BufferPool. My bad.

Reply:
Actually Max is completely right, if the serialise(buf)? returns an error the EncodeBuffer will lose proper alignment for it's read/write pointers. I'll catch the error and make sure the buffer keeps alignment with the next append to the PR.

Reply:
That's...not a good design...shouldn't those pointers be tried to the temporary buffer object you are taking there and be dropped when the buffer itself is dropped at the error?

Reply:
Thanks for the input. I added a `BufferEncoder` which holds a `&mut EncodeBuffer` and impls the BufMut trait for serialisation into the Buffer, it sets `read_offset=write_offset` when the BufferEncoder is dropped.